packages/flutter_ai_providers/lib/src/providers/openwebui.dart

```diff
@@ -0,0 +1,307 @@
+    return _OwuiChatResponse(
+      choices: (json['choices'] as List).map((choice) => _OwuiChatResponseChoice.fromJson(choice)).toList(),
+    );
+  }
+
+  factory _OwuiChatResponse.fromJsonString(String jsonString) {
+    return _OwuiChatResponse.fromJson(jsonDecode(jsonString));
+  }
+}
+
+/// Internal open-webui json encoder / decoder
+/// Decoder for the choice part of [_OwuiChatResponse].
+class _OwuiChatResponseChoice {
+  final ChatMessage message;
+
+  /// Creates an instance of [_OwuiChatResponseChoice].
+  ///
+  /// [message] is the message in the choice.
+  _OwuiChatResponseChoice({required this.message});
+
+  /// Creates an instance of [_OwuiChatResponseChoice] from a JSON object.
+  factory _OwuiChatResponseChoice.fromJson(Map<String, dynamic> json) {
+    return _OwuiChatResponseChoice(
+      message: _OwuiMessage.fromOwuiJson(json),
+    );
+  }
+}
+
+/// Internal open-webui json encoder / decoder
+class _OwuiImageAttachment {
+  final String type;
+  final Map<String, String> imageUrl;
+  final String name;
+
+  _OwuiImageAttachment({
+    required this.type,
+    required this.imageUrl,
+    required this.name,
+  });
+
+  Map<String, dynamic> toJson() => {
+    'type': type,
+    'image_url': imageUrl,
+    'name': name,
+  };
+
+  factory _OwuiImageAttachment.fromImageAttachment(ImageFileAttachment attachment) {
+    final base64Image = base64Encode(attachment.bytes);
+    return _OwuiImageAttachment(
+      type: 'image_url',
+      imageUrl: {'url': "data:${attachment.mimeType};base64,$base64Image"},
+      name: attachment.name,
+    );
+  }
+}
+
+/// Internal open-webui json encoder / decoder
+class _OwuiFileAttachment {
+  final String type;
+  final String id;
+
+  _OwuiFileAttachment({
+    required this.type,
+    required this.id,
+  });
+
+  Map<String, dynamic> toJson() => {
+    'type': type,
+    'id': id,
+  };
+}
+
+/// A provider for [open-webui](https://openwebui.com/)
+/// Use open-webui as unified chat provider.
+class OpenwebuiProvider extends LlmProvider with ChangeNotifier {
+  /// Creates an [OpenwebuiProvider] instance with an optional chat history.
+  ///
+  /// The [history] parameter is an optional iterable of [ChatMessage] objects
+  /// representing the chat history
+  /// The [model] parameter is the ai model to be used for the chat.
+  /// The [baseUrl] parameter is the host of the open-webui server.
+  /// For example port 3000 on localhost use 'http://localhost:3000'
+  /// The [apiKey] parameter is the API key for the open-webui server.
+  /// See the [docs](https://docs.openwebui.com/) for more information.
+  /// Example:
+  /// ``` dart
+  /// LlmChatView(
+  ///   provider: OpenwebuiProvider(
+  ///     host: 'http://127.0.0.1:3000',
+  ///     model: 'llama3.1:latest',
+  ///     apiKey: "YOUR_API_KEY",
+  ///     history: [],
+  ///   ),
+  /// )
+  /// ```
+  OpenwebuiProvider({
+    Iterable<ChatMessage>? history,
+    required String model,
+    required String baseUrl,
```

Comment:
We can default the `baseUrl` param to `http://localhost:3000/api`, which is the default is you run it with the standard config.

Reply:
Done